docs: update 'Caching' page (#1378)
cube-js/cube.js: docs/Cube.js-Backend/Caching.md

Upon incoming request Cube.js first checks the cache using this key.
If nothing is found it executes the query in database, returns result back alongside writing to the cache.

> In case there's an existing value in cache it will be returned only if `refreshKey` value for query isn't changed.

Reviewer: I guess it's quite hard to understand this sentence because it starts with a conditional part and ends with another conditional part. What if we put it like this?

```suggestion
If an existing value is present in the cache and the `refreshKey` value for the query hasn't changed, the cached value will be returned.
```

Reviewer 2: 👍

The `renewQuery` option applies to the `refreshKey` caching system mentioned above, *not* the actual query result cache. If `renewQuery` is passed, Cube.js will always re-execute the `refreshKey` query, skipping that layer of caching, but, if the result of the `refreshKey` query is the same as the last time it ran, that indicates any current query result cache entries are valid, and they will be served. This means that cached data may still be served by Cube.js even if `renewQuery` is passed. This is a good thing: if the underlying data hasn't changed, the expensive query doesn't need to be re-run, and the database doesn't have to work as hard. This does mean that the `refreshKey` SQL must accurately report data freshness for the `renewQuery` to actually work and renew the query.

For situations like real-time analytics or responding to live user changes to underlying data, the `refreshKey` query cache can prevent fresh data from showing up immediately.

> For these situations, the `refreshKey` cache can effectively be disabled by setting the [refreshKey every](cube#parameters-refresh-key) parameter to something very low, like `1 second`.

Author: @paveltiunov @hassankhan I think it is ok to leave the `1 second` suggestion for now, but it is potentially a dangerous road and we need to replace it with streaming techniques.
